### PR TITLE
Add support for DistributedNext

### PR DESCRIPTION
**Pull request by JamesWrigley**
Depends on:
- https://github.com/JuliaData/MemPool.jl/pull/91
- https://github.com/JuliaRegistries/General/pull/120499

### CODE REVIEW

**Comment by JamesWrigley** [comment]
(forgot to commit the new test file :sweat_smile: added in e5160eac)

File: test/preferences.jl [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+
+        cd(dirname(Base.active_project())) do
+            Dagger.set_distributed_package!("Distributed")
+            @test readchomp(cmd) == "Distributed"
+
+            Dagger.set_distributed_package!("DistributedNext")
+            @test readchomp(cmd) == "DistributedNext"
+        end
+    finally
+        set_preferences!(Dagger, "precompile_workload" => false; force=true)
```
Comment (jpsamaroo) [inline test/preferences.jl:19]:
```suggestion
        set_preferences!(Dagger, "precompile_workload" => true; force=true)
```

**Review by jpsamaroo — APPROVED** [review]
Thanks so much!

File: test/preferences.jl (hunk 1 above)
Comment (JamesWrigley) [reply test/preferences.jl]: Good catch :sweat_smile:

**Comment by jpsamaroo** [comment]
Thanks again!

**Pull request merged**